go/detectparser/dhcp.go

```diff
@@ -0,0 +1,39 @@
+package detectparser
+
+import (
+	"fmt"
+	"regexp"
+)
+
+var dhcpRegexPattern1 = regexp.MustCompile(`(DHCPDISCOVER|DHCPOFFER|DHCPREQUEST|DHCPACK|DHCPRELEASE|DHCPINFORM|DHCPEXPIRE) on ([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+) to ([0-9a-f]{2}:[0-9a-f]{2}:[0-9a-f]{2}:[0-9a-f]{2}:[0-9a-f]{2}:[0-9a-f]{2})`)
+
+var dhcpRegexPattern2 = regexp.MustCompile(`(DHCPDISCOVER|DHCPOFFER|DHCPREQUEST|DHCPACK|DHCPRELEASE|DHCPINFORM|DHCPEXPIRE) to ([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+) \(([0-9a-f]{2}:[0-9a-f]{2}:[0-9a-f]{2}:[0-9a-f]{2}:[0-9a-f]{2}:[0-9a-f]{2})\)`)
+
+type DhcpParser struct {
+	Pattern1, Pattern2 *regexp.Regexp
+}
+
+func (s *DhcpParser) Parse(line string) ([]ApiCall, error) {
+	for _, r := range []*regexp.Regexp{s.Pattern1, s.Pattern2} {
+		if matches := r.FindStringSubmatch(line); matches != nil && matches[1] == "DHCPACK" {
+			return []ApiCall{
```

Comment:
Maybe just a check if the ip is really an ip (ParseIP( matches[2]))
Same for the mac address (ParseMAC(matches[3]))

Reply:
Done by the regex.

Reply:
https://i.redditmedia.com/6vYqs23NJ2z8IHqNztxdycCDG9-rtcuGPLpf6nODGxQ.jpg?s=b41acf6471f606d765646ab0c9005094